telethon/helpers.py

```diff
@@ -95,17 +95,20 @@ def strip_text(text, entities):
     return text
 
 
-def retry_range(retries):
+def retry_range(retries, reconnect=False):
     """
     Generates an integer sequence starting from 1. If `retries` is
     not a zero or a positive integer value, the sequence will be
     infinite, otherwise it will end at `retries + 1`.
     """
-    yield 1
+
+    if retries == 0 and not reconnect:
+        yield 1
+
     attempt = 0
     while attempt != retries:
-        attempt += 1
         yield 1 + attempt
+        attempt += 1
```

Comment:
This implementation is wrong: it will yield `1` twice in case of `retry_range(0, False)`.

Reply:
No, it won't, as `attempt != retries` will be false at the first stage and thus never yield an extra 1.

Reply:
Ah, my bad.